src/pages/studyView/table/GeneTable.tsx

```diff
@@ -416,73 +447,40 @@ export class GeneTable extends React.Component<GeneTableProps, {}> {
             (row: GeneTableUserSelectionWithIndex) => row.uniqueKey === uniqueKey
         );
         if (_.isUndefined(record)) {
-            let dataIndex = -1;
-            // definitely there is a match
-            const datum = _.find(
-                this.tableData,
-                (row, index: number) => {
-                    const exist = row.uniqueKey! === uniqueKey;
-                    if (exist) {
-                        dataIndex = index;
-                    }
-                    return exist;
-                }
-            );
-
-            if (!_.isUndefined(datum)) {
-                this.preSelectedRows.push({
-                    rowIndex: dataIndex,
-                    uniqueKey: datum.uniqueKey!,
-                });
-            }
+            this.preSelectedRows.push({ uniqueKey: uniqueKey});
         } else {
-            this.preSelectedRows = _.xorBy(this.preSelectedRows, [record], "rowIndex");
+            this.preSelectedRows = _.xorBy(this.preSelectedRows, [record], "uniqueKey");
         }
     }
 
     @autobind
     @action
     afterSelectingRows() {
+        console.log(JSON.stringify(this.preSelectedRows))
         this.props.onUserSelection(
             this.preSelectedRows.map(row => row.uniqueKey)
         );
         this.preSelectedRows = [];
     }
 
-    @computed
-    get selectedRows() {
-        if (this.props.filters.length === 0) {
-            return [];
-        } else {
-            return _.reduce(
-                this.tableData,
-                (
-                    acc: GeneTableUserSelectionWithIndex[],
-                    row,
-                    index: number
-                ) => {
-                    if (_.includes(this.props.filters, row.uniqueKey)) {
-                        acc.push({
-                            rowIndex: index,
-                            uniqueKey: row.uniqueKey
-                        });
-                    }
-                    return acc;
-                },
-                []
-            );
-        }
-    }
-
     @autobind
     isSelectedRow(data: GeneTableRow) {
         return !_.isUndefined(
-            _.find(_.union(this.selectedRows, this.preSelectedRows), function (row) {
+            _.find(_.union(this.selectedRowsIndexes, this.preSelectedRows), function (row) {
                 return row.uniqueKey === data.uniqueKey;
             })
         );
     }
 
+    @autobind
+    selectedRowClassName(data: GeneTableRow) {
+        const index = _.findIndex(this.props.filters, filter => filter.includes(data.uniqueKey));
+        if (index === -1) {
+            return null;
+        }
+        return index % 2 === 0 ? styles.highlightedEvenRow : styles.highlightedOddRow;
```

Comment:
you don't need to do this with code, you can do it with CSS, for eample in simpleTable/styles.scss we do:
```
tbody>tr:nth-of-type(odd).highlighted {
    background-color: #9FAFD1 !important;
  }

  tbody>tr:nth-of-type(even) { // solid background - hard coded hack, react-bootstrap .table-striped makes ODD # rows grey
    background-color: rgb(255,255,255);
  }

  tbody>tr:nth-of-type(even).highlighted {
    background-color: #B0BED9;
  }
```

Reply:
Here the color is not alternating. alternating color for each selection (there can be more than one row in each selection)

Reply:
ohhh, I see

Reply:
is there a chance of many selected rows? if so then this process will be slow - it would be faster to have a map that takes uniquekey to the first index of filter in which it appears, so that you can check this in constant time